modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectFieldFormBase.tsx

```diff
@@ -297,42 +379,134 @@ export function useObjectFieldForm({
 }
 
 function AttachmentSourceProperty({
+	allowUploadDocAndMedia,
 	disabled,
 	error,
 	objectFieldSettings,
+	objectName,
 	onSettingsChange,
+	setValues,
 }: IAttachmentSourcePropertyProps) {
 	const settings = normalizeFieldSettings(objectFieldSettings);
 
 	const attachmentSource = attachmentSources.find(
 		({value}) => value === settings.fileSource
 	);
 
-	return (
-		<CustomSelect
-			disabled={disabled}
-			error={error}
-			label={Liferay.Language.get('request-files')}
-			onChange={({value}) =>
-				onSettingsChange({
-					name: 'fileSource',
-					value,
-				})
+	const setDocsAndMediaProps = (showFiles: Boolean) => {
+		const updatedSettings: ObjectFieldSetting[] = objectFieldSettings.filter(
+			(setting) => {
+				return (
+					setting.name !== 'showFilesInDocumentsAndMedia' &&
+					setting.name !== 'storageDLFolderPath'
+				);
+			}
+		);
+
+		updatedSettings.push({
+			name: 'showFilesInDocumentsAndMedia',
+			value: showFiles as boolean,
+		});
+
+		if (showFiles) {
+			updatedSettings.push({
+				name: 'storageDLFolderPath',
+				value: `/${objectName}`,
+			});
+		}
+
+		return updatedSettings;
+	};
+
+	useEffect(() => {
```

Comment:
Is there any particular reason for this `useEffect` that we could not handle directly by user's actions?

Reply:
I just used the `useEffect` to set the `ObjectFieldSettings` properly every time the `fileSource` type changes, but  thinking better now, makes more sense to put this logic inside the `<CustomSelect>` component related to the `fileSource` because is the only way to change the `fileSource` prop right? I will make the changes and see if this works fine, thanks!